build/sage_bootstrap/download/mirror_list.py

```diff
@@ -41,17 +41,66 @@ class MirrorListException(RuntimeError):
     pass
         
 
-MIRRORLIST_FILENAME = os.path.join(SAGE_DISTFILES, 'mirror_list')
+class MirrorList(object):
 
+    def __init__(self):
+        self.sources = []
+        upstream_d = os.path.join(SAGE_ROOT, '.upstream.d')
+        for fname in sorted(os.listdir(upstream_d)):
+            if '~' in fname or '#' in fname:
+                # Ignore auto-save and backup files
+                continue
+            try:
+                with open(os.path.join(upstream_d, fname), 'r') as f:
+                    for line in f:
+                        line = line.replace('${SAGE_ROOT}', SAGE_ROOT)
+                        line = line.replace('${SAGE_DISTFILES}', SAGE_DISTFILES)
+                        if '${SAGE_SERVER}' in line:
+                            SAGE_SERVER = os.environ.get("SAGE_SERVER", "")
+                            if not SAGE_SERVER:
+                                continue
+                            line = line.replace('${SAGE_SERVER}',)
+                        line = line.strip()
+                        if line.startswith('#'):
```

Comment:
Speed probably doesn't matter too much, but wouldn't it be a little faster to move `if not line` and `if line.startswith('#')` earlier in this block?

Reply:
Thanks, done in 7e65260